src/format/KdbxXmlReader.cpp

```diff
@@ -1003,7 +983,24 @@ TimeInfo KdbxXmlReader::parseTimes()
 
 QString KdbxXmlReader::readString()
 {
-    return m_xml.readElementText();
+    QXmlStreamAttributes attr = m_xml.attributes();
+    QString value = m_xml.readElementText();
+    bool isProtected = attr.hasAttribute("Protected")
+        && (attr.value("Protected") == "True");
```

Comment:
You should compare case-insensitively and also allow 1 and 0 in addition to True and False.

Reply:
I was wondering and reused the existing code. This seems to be handled differently in different places. Should I introduce a static helper function isTruthy or similar?

Reply:
If you don't have to change too many lines of code, you can do that. Otherwise just handle it locally. KeePass seems to be somewhat inconsistent and sometimes uses True, or true or 1.

Reply:
For reference, KeePass2 does bProtect = ((strProtect != null) && (strProtect == ValTrue)); there. For strict compatibility an equality check may be preferable, but it's unlikely to encounter incompatibilities there.

Reply:
Ye, it dos it here, but while writing the RFC I tested various configurations and I noticed that from other boolean elements/attributes, KeePass2 sometimes writes true/false or 1/0. I therefore made that a feature of the spec. For compatibility we should always write it as True/False, but should support reading true/false and 1/0.